crypto/ml_dsa/ml_dsa_sign.c

```diff
@@ -126,8 +129,10 @@ int ossl_ml_dsa_mu_update(EVP_MD_CTX *md_ctx, const uint8_t *msg, size_t msg_len
  */
 int ossl_ml_dsa_mu_finalize(EVP_MD_CTX *md_ctx, uint8_t *mu, size_t mu_len)
 {
-    if (mu_len != ML_DSA_MU_BYTES)
+    if (!ossl_assert(mu_len != ML_DSA_MU_BYTES)) {
```

Comment:
You need to invert this condition as there must be negation before the ossl_assert() call.